apps/admin-gui/src/app/users/pages/user-detail-page/user-settings/user-roles/user-roles.component.ts

```diff
@@ -44,21 +44,51 @@ export class UserRolesComponent implements OnInit {
 
   getData(): void {
     this.outerLoading = true;
+
+    // get user based roles
     this.roles.clear();
-    of(this.entityType)
-      .pipe(
-        mergeMap((type) =>
-          iif(
-            () => type === 'SELF',
-            of(this.store.getPerunPrincipal().roles),
-            this.authzResolverService.getUserRoles(this.userId)
-          )
-        )
-      )
-      .subscribe((roles) => {
+    this.authzResolverService.getUserDirectRoles(this.userId).subscribe(
+      (roles) => {
+        // prepare direct roles
         const roleNames = Object.keys(roles).map((role) => role.toUpperCase());
         this.roles = this.roleService.prepareRoles(roles, roleNames);
-        this.outerLoading = false;
-      });
+
+        // get group based roles
+        this.indirectRoles.clear();
+        this.authzResolverService
+          .getUserRolesObtainedFromAuthorizedGroupMemberships(this.userId)
+          .subscribe(
+            (groupBasedRoles) => {
+              // prepare group based roles
+              const groupBasedRoleNames = Object.keys(groupBasedRoles).map((role) =>
+                role.toUpperCase()
+              );
+              this.indirectRoles = this.roleService.prepareRoles(
+                groupBasedRoles,
+                groupBasedRoleNames
+              );
+
+              // get map of roles and complementary objects with authorized groups
+              this.rolesComplementaryObjectsWithAuthzGroups.clear();
+              this.authzResolverService
+                .getRoleComplementaryObjectsWithAuthorizedGroups(this.userId)
+                .subscribe(
+                  (obtainedComplementaryObjects) => {
+                    // prepare map
+                    this.rolesComplementaryObjectsWithAuthzGroups =
+                      this.roleService.prepareComplementaryObjects(
+                        Object.keys(obtainedComplementaryObjects),
+                        obtainedComplementaryObjects
+                      );
+                    this.outerLoading = false;
+                  },
+                  () => (this.outerLoading = false)
+                );
```

Comment:
```suggestion
                .subscribe({
                  next: (obtainedComplementaryObjects) => {
                    // prepare map
                    this.rolesComplementaryObjectsWithAuthzGroups =
                      this.roleService.prepareComplementaryObjects(
                        Object.keys(obtainedComplementaryObjects),
                        obtainedComplementaryObjects
                      );
                    this.outerLoading = false;
                  },
                  error: () => (this.outerLoading = false)
                });
```
use the non-deprecated syntax with all the subscribe methods please (like this I believe)